packages/v8-deopt-viewer/src/index.js

```diff
@@ -110,11 +111,24 @@ export default async function run(srcFile, options) {
 	await mkdir(options.out, { recursive: true });
 
 	console.log("Parsing log...");
-	const logContents = await readFile(logFilePath, "utf8");
+
+	const fd = await openFile(logFilePath);
+	const { buffer: logContentsSlice } = await fd.read({length: 64 * 1024 * 1024});
+	await fd.close();
 
 	// New IC format has 10 values instead of 9
-	const hasNewIcFormat = /\w+IC(,.*){10}/.test(logContents);
-	const rawDeoptInfo = await parseV8Log(logContents, {
+	// todo parse first line - v8-version,8,4,371,19,-node.18,0, instead 
+	// https://github.com/andrewiggins/v8-deopt-viewer/issues/47
+	const hasNewIcFormat = /\w+IC(,.*){10}/.test(logContentsSlice.toString());
```

Comment:
I think the best place for it is somewhere in the `DeoptLogReader`. It processes all the log lines anyways, therefore it can inspect them until the proper format is detected. And there will be no need for additional read of 64MB buffer from the log file and match it against IC pattern.

Something like this:

```js
processLogLine_(line) {
    if (this.hasNewIcFormat === undefined && /^\w+IC,/.test(line)) {
        this.hasNewIcFormat = /^\w+IC(,.*){10}/.test(line);
        this.updateDispatchTableForIc();
    }
}

updateDispatchTableForIc() {
    const processIc = this.hasNewIcFormat ? ...;
    const parseIcField = this.hasNewIcFormat ? ...
    this.dispatchTable_.LoadIC = ...;
    ...
}
```